client/src/pages/admin/pages/add-data-source.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { useState, useRef  } from 'react';
+import { DataSource } from '@/api/Models';
+import { useAddDataSource } from '@/api/datasource/queries';
+import { ClientType } from '@/api/Models';
+
+
+export const Form = () => {
+      const addDataSource = useAddDataSource()
+      const formRef = useRef<HTMLFormElement>(null);
+
+      const handleSubmit = (event: React.FormEvent) => {
+        event.preventDefault();
+        const form = formRef.current;
+
+        const formData: DataSource = {
+          type: ClientType.API,
+          name: event.target.elements.name.value,
+          account: event.target.elements.account.value,
+          container: event.target.elements.container.value,
+          description: event.target.elements.description.value,
+          imageURL: event.target.elements.imageURL.value,
+          sasToken: event.target.elements.sasToken.value
+        }
+        addDataSource.mutate(formData)
+        form.reset();
+
+      };
+
+      return (
```

Comment:
I thought we were doing a modal?

Reply:
Planning to do a modal in a different PR with the modal extraction to stop this PR from getting too messy